Change: Try stopping extmidi player with SIGINT first.

PeterN opened the pull request:
## Motivation / Problem

On Linux with extmidi music playback, stopping playback or switching to the next track may leave stuck notes. This is because it's down to the external player to turn those notes off, and we kill the process with SIGTERM which can interrupt this.

<!--
Describe here shortly
* For bug fixes:
    * What problem does this solve?
    * If there is already an issue, link the issue, otherwise describe the problem here.
* For features or gameplay changes:
    * What was the motivation to develop this feature?
    * Does this address any problem with the gameplay or interface?
    * Which group of players do you think would enjoy this feature?
-->


## Description

Instead, try sending SIGINT first, and if that fails fall back to SIGTERM then SIGKILL. This can give the player a chance to issue MIDI note-off commands.

This fixes at least pmidi leaving stuck notes on.

<!--
Describe here shortly
* For bug fixes:
    * How is the problem solved?
* For features or gameplay changes:
    * What does this feature do?
    * How does it improve/solve the situation described under 'motivation'.
-->


## Limitations

<!--
Describe here
* Is the problem solved in all scenarios?
* Is this feature complete? Are there things that could be added in the future?
* Are there things that are intentionally left out?
* Do you know of a bug or corner case that does not work?
-->


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

rubidium42 commented on src/music/extmidi.cpp:
```diff
@@ -130,23 +130,32 @@ void MusicDriver_ExtMidi::DoPlay()
+	if (this->pid <= 0) return;
+
+	/* First try to gracefully stop for about five seconds;
+	 * 5 seconds = 5000 milliseconds, 10 ms per cycle => 500 cycles. */
```
Not sure how to solve this exactly, but... with this change the magic 500 and 10 in `DoKillWait` are not documented in the function, but rather here even though they are essentially implementation details of how to get to the 5 seconds. On the other hand, writing down it is going to gracefully try to interrupt, then terminate and if all else fails kill the driver belongs to this function. So I guess the comment ought to be split over the two functions.

PeterN committed:
Change: Try stopping extmidi player with SIGINT first.

This may give the player a chance to issue MIDI note-off commands.

The kill/waitpid cycle is also less aggressive, waiting 50ms each time.
